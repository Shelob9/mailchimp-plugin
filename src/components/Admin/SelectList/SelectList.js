import React, {Fragment} from 'react'
import {HorizontalForm} from '@calderajs/forms';

/**
 * Select a MailChimp List
 *
 * @param listFieldConfig
 * @param listId
 * @param setList
 * @param instanceId
 * @return {*}
 * @constructor
 */
export const SelectList = ({listFieldConfig,listId,setList,instanceId}) => (
	<Fragment>
		<HorizontalForm
			fields={[listFieldConfig]}
			initialValues={
				{
					[listFieldConfig.fieldId]:listId
				}
			}
			onChange={(values)=>{
				setList(values[listFieldConfig.fieldId]);
			}}
			instanceId={`caldera-mc-select-${instanceId}` }
		/>
	</Fragment>
);
