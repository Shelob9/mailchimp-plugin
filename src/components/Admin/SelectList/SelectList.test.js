import React from 'react';
import {mount} from 'enzyme';
import renderer from 'react-test-renderer';
import {SelectList} from "./SelectList";

const field = [{"fieldType":"select","required":true,"fieldId":"mc-select-field","options":[{"value":"45907f0c59","label":"Future Capable"}]}];
it( 'Shows list', () => {
	expect( renderer.create(
		<SelectList
			listFieldConfig={field}
			listId={'45907f0c59'}
			setList={()=>{}}
			instanceId={'a'}
		/>
	) ).toMatchSnapshot();
});

it( 'Changes list', () => {
	const onChange = jest.fn();
	const component =  mount(
		<SelectList
			listFieldConfig={field[0]}
			listId={''}
			setList={onChange}
			instanceId={'1'}
		/>
	);

	component.find( 'select' ).first().simulate(
		'change',
		{target: {value:'45907f0c59'}}
	);
	expect( onChange.mock.calls.length).toBe(1)

});
