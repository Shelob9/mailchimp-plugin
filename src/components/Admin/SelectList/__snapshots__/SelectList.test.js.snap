// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shows list 1`] = `
<div
  className="caldera"
>
  <form
    className="horizontal-form-caldera-mc-select-a caldera-form"
    onSubmit={[Function]}
  >
    <div
      className="sc-bdVaJa caldera-form-row caldera-row sc-bwzfXH pjOvZ"
    >
      <div
        className="caldera-column width-1 sc-bdVaJa iHZvIS"
        id="c-caldera-mc-select-a-0"
      />
    </div>
  </form>
</div>
`;
